src/material/button/button.md

```diff
@@ -33,10 +33,46 @@ screen-readers, which will read the text character-by-character. We leave the de
 approach this to the consuming app.
 
 ### Accessibility
-Angular Material uses native `<button>` and `<a>` elements to ensure an accessible experience by
-default. The `<button>` element should be used for any interaction that _performs an action on the
-current page_. The `<a>` element should be used for any interaction that _navigates to another
-view_.
 
+The ARIA spec gives different uses for buttons and links. In order to give the correct information to
+assistive technology, it is important to use the correct one for each situation. 
+
+ - Use the `<button>` element for any interactions that _performs an action on the
+current page_.
+ - Use the `<a>` element for links. That is iteractions that _navigates to another
+view_. This includes navigating to another page, another view in a single-page application and to elements on the same page. Include the href attribute when using the `<a>` element.
+
+#### Links That Look Like Buttons
+Generally, the `<a>` should be used for interactions that perform navigation. It ensures that assistive technology has the
+correct semantic information for the interaction. If a `<button>` must be used, then do the following.
+
+ - Give it the `role='link'` attribute.
+ - Assign an `href` attribute with the link's destination.
+
+#### Icon Buttons
 Buttons or links containing only icons (such as `mat-fab`, `mat-mini-fab`, and `mat-icon-button`)
 should be given a meaningful label via `aria-label` or `aria-labelledby`.
+
+As always, images inside buttons should contain an meaningful alt text.
```

Comment:
This is really only necessary if the image isn't purely decorative. For example, consider the following example:

```
<button>
  <img src="search-icon.png" alt="">
  Search
</button>
```

In this case, it's reasonable to hide the search icon/image from SRs with `alt=""` because it's "fully described" by the adjacent text "Search" (https://www.w3.org/WAI/tutorials/images/decorative/).

Reply:
I would go even further to say that people will almost _never_ put meaningful image content inside of a `mat-button` (people might wrap images in an `<a>`, but that's not related to use of `MatButton`). 

For icon buttons, I would link to the guidance for the icon component, since I don't want to repeat that information in two places.

Reply:
Yeah, you're right per https://www.w3.org/TR/wai-aria-practices-1.1/#children_presentational (of which `button` is one of the roles).